File: app/src/main/java/com/antyzero/smoksmog/logger/Logger.java
package com.antyzero.smoksmog.logger;


public interface Logger {

    // Verbose

    void v( String tag, String message );

    void v( String tag, String message, Throwable throwable );

    // Debug

    void d( String tag, String message );

    void d( String tag, String message, Throwable throwable );

    // Info

    void i( String tag, String message );

    void i( String tag, String message, Throwable throwable );

    // Warning

    void w( String tag, String message );

    void w( String tag, String message, Throwable throwable );

    // Error

    void e( String tag, String message );

    void e( String tag, String message, Throwable throwable );
}

File: android/src/main/kotlin/smoksmog/logger/Logger.kt
package smoksmog.logger


interface Logger {

    // Verbose

    fun v(tag: String, message: String)

    fun v(tag: String, message: String, throwable: Throwable)

    // Debug

    fun d(tag: String, message: String)

    fun d(tag: String, message: String, throwable: Throwable)

    // Info

    fun i(tag: String, message: String)

    fun i(tag: String, message: String, throwable: Throwable)

    // Warning

    fun w(tag: String, message: String)

    fun w(tag: String, message: String, throwable: Throwable)

    // Error

    fun e(tag: String, message: String)

    fun e(tag: String, message: String, throwable: Throwable)
}
